From 50b6a55886b1c737ece44c0487d4e1e1d27c7b10 Mon Sep 17 00:00:00 2001
From: Cheng Jian <cj.chengjian@huawei.com>
Date: Mon, 15 Nov 2021 19:35:43 +0800
Subject: [PATCH] sched/fair: fix sd_llc_alloc_all() compile error
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 50b6a55886b1c737ece44c0487d4e1e1d27c7b10

MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

hulk inclusion
category: bugfix
bugzilla: 38261, https://gitee.com/openeuler/kernel/issues/I49XPZ
CVE: NA

---------------------------

when CONFIG_SCHED_STEAL disabled:
	kernel/sched/topology.c:24:74: warning: ‘struct s_data’ declared inside parameter list will not be visible outside of this definition or declaration
	  24 | static inline int sd_llc_alloc_all(const struct cpumask *cpu_map, struct s_data *d) { return 0; }
	     |                                                                          ^~~~~~

	kernel/sched/topology.c: In function ‘build_sched_domains’:
	kernel/sched/topology.c:2188:32: error: passing argument 2 of ‘sd_llc_alloc_all’ from incompatible pointer type [-Werror=incompatible-pointer-types]
	2188 |  if (sd_llc_alloc_all(cpu_map, &d))
	     |                                ^~
             |                                |
             |                                struct s_data *

Signed-off-by: Cheng Jian <cj.chengjian@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 kernel/sched/topology.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/sched/topology.c b/kernel/sched/topology.c
index 0564aeabbcb8..fcf6aebb13c4 100644
--- a/kernel/sched/topology.c
+++ b/kernel/sched/topology.c
@@ -13,8 +13,8 @@ DEFINE_MUTEX(sched_domains_mutex);
 static cpumask_var_t sched_domains_tmpmask;
 static cpumask_var_t sched_domains_tmpmask2;
 
-#ifdef CONFIG_SCHED_STEAL
 struct s_data;
+#ifdef CONFIG_SCHED_STEAL
 static int sd_llc_alloc(struct sched_domain *sd);
 static void sd_llc_free(struct sched_domain *sd);
 static int sd_llc_alloc_all(const struct cpumask *cpu_map, struct s_data *d);
-- 
2.26.2

